src/vector/backends/awkward_constructors.py

```diff
@@ -311,7 +311,17 @@ def Array(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
     import vector
     import vector.backends.awkward
 
-    akarray = awkward.Array(*args, **kwargs)
+    ak_array_types: tuple[typing.Any, ...] = (awkward.Array,)
+
+    if vector._is_awkward_v2:
+        import dask_awkward  # type: ignore[import-not-found]
+
+        ak_array_types += (dask_awkward.Array,)
```

Comment:
This is an example of why I think we should drop Awkward 1 support soon. Generally, the typing code should be declarative.

By the way, I think you can declare a `dask_awkward.Array` dependency without importing `dask_awkward` by using a string: `"dask_awkward.Array"`. I'm not 100% sure about typing syntax; maybe this only works past a particular Python version. However, we _want_ to be able to allow some types without necessarily importing them (a start-up time cost) or even having them installed (a dependency-complexity cost).

Reply:
I see, yes, that makes sense. I could remove awkward v1 support in this PR itself, but it might be better to do that in the `v1.3.1`/`v1.4.0` release. [PEP 484 mentions forward referencing](https://peps.python.org/pep-0484/#forward-references) to deal with such cases, but that will only be helpful while adding type hints.

I have pushed a commit rewriting the logic for handling `dask_awkward` arrays. The implementation does not require importing `dask_awkward`. Please let me know if that works!

Reply:
Dropping a major dependency version should happen at some significant version threshold in Vector. Maybe 1.3.x → 1.4.0, or maybe 1.x → 2.0.0.

I like the idea of 2.0.0 (Uproot increases major version when Awkward increases major version), but this has scared users in the past: seeing Uproot 4 → 5 made some people think that the changes in Uproot were major, rather than the changes in its dependencies.

So maybe 1.4.0, yeah.

Reply:
Thanks! I'll drop awkward v1 in the next release.